public/app/features/dashboard/utils/getPanelMenu.ts

```diff
@@ -147,6 +164,90 @@ export function getPanelMenu(
     shortcut: 'p s',
   });
 
+  const exportMenu: PanelMenuItem[] = [];
+
+  // geomap, logs, and news panels not currently supported by image export
+  if (!(['geomap', 'logs', 'news'].indexOf(panel.type) > -1)) {
+    let exportImageMenu = [];
+
+    exportImageMenu.push({
+      text: t('panel.header-menu.export-png', `PNG`),
+      onClick: () => onExportPanel(ExportType.png),
+    });
+
+    exportImageMenu.push({
+      text: t('panel.header-menu.export-jpeg', `JPEG`),
+      onClick: () => onExportPanel(ExportType.jpeg),
+    });
+
+    exportImageMenu.push({
+      text: t('panel.header-menu.export-bmp', `BMP`),
+      onClick: () => onExportPanel(ExportType.bmp),
+    });
+
+    exportMenu.push({
+      type: 'submenu',
+      text: `Image`,
+      iconClassName: 'image-v',
+      subMenu: exportImageMenu,
+    });
+  }
+
+  if (panel.plugin && !panel.plugin.meta.skipDataQuery) {
+    let exportDataMenu = [];
+
+    exportDataMenu.push({
+      text: t('panel.header-menu.export-csv', `CSV`),
+      onClick: () => onExportPanel(ExportType.csv),
+    });
+
+    exportDataMenu.push({
+      text: t('panel.header-menu.export-excel', `Excel`),
+      onClick: () => onExportPanel(ExportType.xlsx),
+    });
+
+    exportDataMenu.push({
+      text: t('panel.header-menu.export-json', `Panel JSON`),
+      onClick: () => onExportPanel(ExportType.panelJson),
+    });
+
+    exportDataMenu.push({
+      text: t('panel.header-menu.export-data', `Data JSON`),
+      onClick: () => onExportPanel(ExportType.dataJson),
+    });
+
+    exportDataMenu.push({
+      text: t('panel.header-menu.export-dataframe', `DataFrame JSON`),
+      onClick: () => onExportPanel(ExportType.dataFrameJson),
+    });
+
+    exportMenu.push({
+      type: 'submenu',
+      text: `Data`,
+      iconClassName: 'book',
+      subMenu: exportDataMenu,
+    });
+  } else {
+    exportMenu.push({
+      text: t('panel.header-menu.export-json', `Panel JSON`),
+      iconClassName: 'book',
+      onClick: () => onExportPanel(ExportType.panelJson),
+    });
+  }
+
+  // TODO: Re-implement real feature toggle
+
+  const featureToggleEnabled = true;
```

Comment:
Looks like this still needs to be implemented, I think

Reply:
Got that sorted now 👍